src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs

```diff
@@ -0,0 +1,296 @@
+                {
+                    if (nameArityMap.TryGetValue(symbol.Name, out var arities))
+                    {
+                        arities.Add(symbol.GetArity());
+                    }
+                    else
+                    {
+                        nameArityMap[symbol.Name] = new List<int>() { symbol.GetArity() };
+                    }
+                }
+
+                foreach (var name in nameArityMap.Keys)
+                {
+                    foreach (var memberWithName in namespaceSymbol.GetMembers(name))
+                    {
+                        if (nameArityMap.TryGetValue(memberWithName.Name, out var arities) && arities.Contains(memberWithName.GetArity()))
+                        {
+                            return true;
+                        }
+                    };
+                }
+
+                return false;
+            }
+
+            private static bool TryGetQualifiedNamespaceWithOutConflict(Compilation compilation, string qualifiedNamespaceName, out INamespaceSymbol namespaceSymbol)
+            {
+                namespaceSymbol = compilation.GlobalNamespace;
+                if (qualifiedNamespaceName == string.Empty)
+                {
+                    return true;
+                }
+
+                foreach (var name in qualifiedNamespaceName.Split('.'))
+                {
+                    var membersWithName = namespaceSymbol.GetMembers(name);
+                    if (membersWithName.Any())
+                    {
+                        foreach (var member in membersWithName)
+                        {
+                            switch (member)
+                            {
+                                case INamespaceSymbol namespaceMember:
+                                    namespaceSymbol = namespaceMember;
+                                    break;
+                                case INamedTypeSymbol typeMember:
+                                    if (typeMember.Arity == 0)
+                                    {
+                                        // The namespace confict with an existing type declaration.
+                                        namespaceSymbol = null;
+                                        return false;
+                                    }
+                                    break;
+                                default:
+                                    Debug.Assert(false, member.Kind.ToString());
+                                    break;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        namespaceSymbol = null;
+                    }
+
+                    // No type conflict but matcing namespace is not declared.
+                    if (namespaceSymbol == null)
+                    {
+                        break;
+                    }
+                }
+                return true;
+            }
```

Comment:
skipping this until we can have a discussion aobut conflicts and what we will/won't check for.